src/components/tasks/TaskSearch/Suggestion/Pill.tsx

```diff
@@ -0,0 +1,95 @@
+import { TaskSearchOption } from '@/interfaces/searchOptions.type';
+import { useRef } from 'react';
+import classNames from './pill.module.scss';
+
+export interface PillProps {
+    idx: number;
+    option: TaskSearchOption;
+    newPillValue: string;
+    handleKeyPress: (event: React.KeyboardEvent<HTMLInputElement>) => void;
+    setNewPillValue: (input: string) => void;
+    removePill: (idx: number) => void;
+    selectedPill: false | number;
+    pillToBeRemoved: number;
+    setSelectedPill: (key: number | false) => void;
+}
+
+export default function RenderPills({
+    idx,
+    option,
+    newPillValue,
+    setNewPillValue,
+    handleKeyPress,
+    removePill,
+    selectedPill,
+    pillToBeRemoved,
+    setSelectedPill,
+}: PillProps) {
+    const [field] = Object.keys(option);
+    const value = option[field];
+    const pillInputRef = useRef<HTMLInputElement>(null);
+    if (selectedPill === idx) {
+        pillInputRef.current?.focus();
+    }
+    const activatePillInEditMode = () => {
+        setNewPillValue(`${field}:${value}`);
+        setSelectedPill(idx);
+    };
+    return (
+        <>
+            {selectedPill === idx ? (
+                <div
+                    style={{
+                        width: `${newPillValue.length * 0.95}ch`,
+                    }}
+                    className={classNames['parent-pill']}
+                >
+                    <input
+                        data-testid="pill-input"
+                        ref={pillInputRef}
+                        onChange={(e) => {
+                            setNewPillValue(e.target.value);
+                            e.preventDefault();
+                        }}
+                        onKeyDown={handleKeyPress}
+                        defaultValue={`${field}:${value}`}
+                        className={classNames['pill-input']}
+                    />
+                </div>
+            ) : (
+                <div
+                    data-testid="parent-pill"
+                    className={`parent-pill ${classNames['parent-pill']} ${
+                        classNames['selected-options']
+                    } ${pillToBeRemoved == idx ? classNames['highlight'] : ''}`}
+                >
+                    <span
+                        data-testid="pill-content"
+                        onClick={activatePillInEditMode}
+                        className={classNames['pill-content']}
+                    >
+                        {' '}
+                        {`${field}:${value}`}
```

Comment:
nit: The variable names here are very generic.

Reply:
Yes, but I don't think this will cause any issues because [here](https://github.com/Real-Dev-Squad/website-status/blob/80bc8e4f1981227ac0d42b8a28a46381f72f1c1e/src/components/tasks/TaskSearch/Suggestion/Pill.tsx#L28) it is quite understandable what does field and value represent